src/promptflow/promptflow/_sdk/_service/apis/telemetry.py

```diff
@@ -0,0 +1,148 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from flask import jsonify, make_response, request
+
+from promptflow._sdk._service import Namespace, Resource
+from promptflow._sdk._service.utils.utils import build_pfs_user_agent, local_user_only
+from promptflow._utils.utils import camel_to_snake
+from promptflow.exceptions import UserErrorException
+
+api = Namespace("Telemetries", description="Telemetry Management")
+
+dict_field = api.schema_model("RunDict", {"additionalProperties": True, "type": "object"})
+
+
+class EventType:
+    START = "Start"
+    END = "End"
+
+
+class AllowedActivityName:
+    FLOW_TEST = "pf.flow.test"
+    FLOW_NODE_TEST = "pf.flow.node_test"
+    GENERATE_TOOL_META = "pf.flow._generate_tools_meta"
+
+
+def _dict_camel_to_snake(data):
+    if isinstance(data, dict):
+        result = {}
+        for key, value in data.items():
+            result[camel_to_snake(key)] = _dict_camel_to_snake(value)
+        return result
+    else:
+        return data
+
+
+def parse_activity_info(telemetry_data, user_agent):
+    from promptflow._sdk._telemetry.activity import generate_request_id
+
+    first_call = telemetry_data.get("firstCall", True)
+    request_id = telemetry_data.get("requestId", generate_request_id())
+
+    return {
+        "request_id": request_id,
+        "first_call": first_call,
+        "user_agent": user_agent,
+        **_dict_camel_to_snake(telemetry_data.get("metadata", {})),
+    }
+
+
+def validate_telemetry_payload(payload):
+    from promptflow._sdk._telemetry import ActivityCompletionStatus, ActivityType
+
+    if not all(key in payload for key in ("eventType", "timestamp", "metadata")):
+        missing_fields = {"eventType", "timestamp", "metadata"} - set(payload.keys())
+        raise UserErrorException(f"Missing required fields in telemetry payload: {', '.join(missing_fields)}")
+    if payload.get("eventType") not in (EventType.START, EventType.END):
+        raise UserErrorException(f"Invalid eventType: {payload.get('eventType')}")
+    metadata = payload.get("metadata")
+    if not isinstance(metadata, dict):
+        raise UserErrorException(f"Invalid metadata: {metadata}")
+    if metadata.get("activityName") not in [
+        AllowedActivityName.FLOW_TEST,
+        AllowedActivityName.FLOW_NODE_TEST,
+        AllowedActivityName.GENERATE_TOOL_META,
+    ]:
+        raise UserErrorException(f"Invalid activityName: {metadata.get('activityName')}")
+    if metadata.get("activityType") not in [
+        ActivityType.INTERNALCALL,
+        ActivityType.PUBLICAPI,
+    ]:
+        raise UserErrorException(f"Invalid activityType: {metadata.get('activityType')}")
```

Comment:
Can add `error_target `and `message_format`?

Reply:
C# can set them but they're optional